spinalcordtoolbox/scripts/sct_analyze_lesion.py

```diff
@@ -311,6 +325,210 @@ def _measure_axial_damage_ratio(self, im_data, p_lst, idx):
+        tissue_bridges_dict = {}
+        # Loop across sagittal slices
+        for sagittal_slice in sagittal_lesion_slices:
+            # Get all axial slices (S-I direction) with the lesion for the selected sagittal slice
+            # In other words, we will iterate through the lesion in S-I direction and compute tissue bridges for each
+            # axial slice with the lesion
+            axial_lesion_slices = np.unique(np.where(im_lesion_data[sagittal_slice, :, :])[1])     # as the orientation is RPI, [1] is the S-I direction
+            # Iterate across axial slices to compute tissue bridges
+            for axial_slice in axial_lesion_slices:
+                # Get the lesion segmentation mask of the selected 2D axial slice
+                slice_lesion_data = im_lesion_data[sagittal_slice, :, axial_slice]
+                # Get the spinal cord segmentation mask of the selected 2D axial slice
+                slice_sc_data = im_sc_data[sagittal_slice, :, axial_slice]
+                # Get the indices of the lesion mask for the selected axial slice
+                lesion_indices = np.where(slice_lesion_data)[0]
+                # Get the indices of the spinal cord mask for the selected axial slice
+                sc_indices = np.where(slice_sc_data)[0]
+
+                # Compute ventral and dorsal tissue bridges
+                dorsal_bridge_width = lesion_indices[0] - sc_indices[0]         # [0] returns the most dorsal elements
+                # if the lesion extends the spinal cord, the dorsal bridge is set to 0
+                if dorsal_bridge_width < 0:
+                    dorsal_bridge_width = 0
+                ventral_bridge_width = sc_indices[-1] - lesion_indices[-1]      # [-1] returns the most ventral elements
+                # if the lesion extends the spinal cord, the ventral bridge is set to 0
+                if ventral_bridge_width < 0:
+                    ventral_bridge_width = 0
+
+                tissue_bridges_dict[sagittal_slice, axial_slice] = \
+                    {'dorsal_bridge_width': dorsal_bridge_width,
+                     'ventral_bridge_width': ventral_bridge_width,
+                     'lesion_indices': lesion_indices}       # we store lesion indices for plotting
+
+        # --------------------------------------
+        # Get minimal tissue bridges
+        # --------------------------------------
+        # Convert the dictionary to a DataFrame (for easier manipulation)
+        # 1. Create a MultiIndex from the dictionary keys
+        index = pd.MultiIndex.from_tuples(tissue_bridges_dict.keys(), names=['sagittal_slice', 'axial_slice'])
+        # 2. Create the DataFrame using the MultiIndex and the dictionary values
+        tissue_bridges_df = pd.DataFrame(list(tissue_bridges_dict.values()), index=index)
+        # 3. Reset the index to make 'sagittal_slice' and 'axial_slice' as columns
+        tissue_bridges_df.reset_index(inplace=True)
```

Comment:
I think that you could also keep the `MultiIndex` to simplify indexing here too?

For example:

```python
# before
df_temp = tissue_bridges_df[tissue_bridges_df['sagittal_slice'] == sagittal_slice]
# after
df_temp = tissue_bridges_df[slice(sagittal_slice)]
```

Slicing MultiIndexes is [a bit more advanced](https://pandas.pydata.org/docs/user_guide/advanced.html#using-slicers), but it might be useful to work through as a way of learning `pandas` a little better?

Reply:
Interesting! I tried keeping the `MultiIndex`, but the code became a little bit less readable, tbh. Maybe I would keep indexing as it is for now.